include/zephyr/init.h

```diff
@@ -207,7 +207,9 @@ struct init_entry {
 		Z_INIT_ENTRY_SECTION(level, prio, 0) __used __noasan           \
 		Z_INIT_ENTRY_NAME(name) = {                                    \
 			.init_fn = {.sys = (init_fn_)},                        \
-			.dev = NULL,                                           \
+			{                                                      \
+				.dev = NULL,                                   \
+			},                                                     \
```

Comment:
tempted to just drop this entirely, it's zero initialized anyway

Reply:
fine with dropping

Reply:
dropped an inlined